File: app/src/main/java/com/wakwak/keyboardshowhidesample/MainActivity.java
package com.wakwak.keyboardshowhidesample;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewTreeObserver;
import android.view.Window;
import android.view.WindowManager;
import android.widget.EditText;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = MainActivity.class.getSimpleName();
    ImeUtil imeUtil;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        toAdjustResize();
        imeUtil = ImeUtil.get();
        ImeUtil.set(imeUtil);
        final ImeDetectFrameLayout rootView = (ImeDetectFrameLayout) findViewById(R.id.ime_detact_frame_layout);
        final ImageView attachmentButton = (ImageView) findViewById(R.id.button_plus);
        final View pickerContainer = findViewById(R.id.picker_container);
        final EditText editText = (EditText) findViewById(R.id.editText);
        attachmentButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (pickerContainer.getVisibility() != View.VISIBLE) {
                    WindowManager.LayoutParams params = getWindow().getAttributes();
                    imeUtil.hideImeKeyboard(getApplicationContext(), editText);
                    toAdjustNothing();
                    pickerContainer.setVisibility(View.VISIBLE);
                    getWindow().setAttributes(params);
                } else {
                    pickerContainer.setVisibility(View.GONE);
                    toAdjustResize();
                    imeUtil.showImeKeyboard(getApplicationContext(), editText);
                }
            }
        });

        editText.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                if (event.getAction() == MotionEvent.ACTION_DOWN) {
                    toAdjustResize();
                    if (pickerContainer.getVisibility() == View.VISIBLE) {
                        pickerContainer.setVisibility(View.GONE);
                    }
                    imeUtil.showImeKeyboard(getApplicationContext(), editText);
                    return true;
                }
                return false;
            }
        });
    }

    @Override
    protected void onDestroy() {
        ImeUtil.clearInstance();
        super.onDestroy();
    }

    private void toAdjustNothing() {
        getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_NOTHING);
    }

    private void toAdjustResize() {
        getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);
    }
}

File: app/src/main/java/com/wakwak/keyboardshowhidesample/MainActivity.kt
package com.wakwak.keyboardshowhidesample

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.MotionEvent
import android.view.View
import android.widget.EditText
import android.widget.ImageView
import android.widget.RelativeLayout
import com.wakwak.keyboardshowhidesample.extension.gone
import com.wakwak.keyboardshowhidesample.extension.isVisible
import com.wakwak.keyboardshowhidesample.extension.visible

class MainActivity : AppCompatActivity() {

    @SuppressLint("ClickableViewAccessibility")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //toAdjustResize()

        val attachmentButton = findViewById<ImageView>(R.id.button_plus)
        val pickerContainer = findViewById<RelativeLayout>(R.id.picker_container)
        val editText = findViewById<EditText>(R.id.editText)

        attachmentButton.setOnClickListener {
            if (!pickerContainer.isVisible()) {
                val params = window.attributes
                ImeUtil.hideImeKeyboard(this, editText)
                //toAdjustNothing()
                pickerContainer.visible()
                window.attributes = params
            } else {
                pickerContainer.gone()
                //toAdjustResize()
                ImeUtil.showImeKeyboard(this, editText)
            }
        }

        editText.setOnTouchListener(View.OnTouchListener { _, event ->
            if (event.action == MotionEvent.ACTION_DOWN) {
                //toAdjustResize()
                if (pickerContainer.isVisible()) {
                    pickerContainer.gone()
                }
                ImeUtil.showImeKeyboard(this, editText)
                return@OnTouchListener true
            }
            false
        })
    }
}
